lib/_stream_readable.js

```diff
@@ -543,10 +543,16 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
   // is attached before any userland ones.  NEVER DO THIS.
   if (!dest._events || !dest._events.error)
     dest.on('error', onerror);
-  else if (Array.isArray(dest._events.error))
-    dest._events.error.unshift(onerror);
-  else
-    dest._events.error = [onerror, dest._events.error];
+  else {
+    var existingHandler = dest._events.error;
+    if (Array.isArray(existingHandler))
+      existingHandler.unshift(onerror);
+    else {
+      var handlers = [onerror, existingHandler];
+      handlers.onceCount = (existingHandler.once ? 1 : 0);
+      dest._events.error = handlers;
+    }
+  }
```

Comment:
I think this *might* pose a problem – Readable and Duplex streams coming out of pegged versions of `readable-stream` on npm won't have this logic, and so won't preserve the `once`-ness of existing error handlers after these revisions.

For example:

```javascript
var Readable = require('readable-stream').Readable;
var rs = new Readable;

var Writable = require('stream').Writable;
var ws = new Writable;
writable.once('error', function() {
  // important code to only perform once
});
rs.pipe(ws);
```